lib/drizzlepac/catalogs.py

```diff
@@ -382,8 +382,9 @@ def plotXYCatalog(self, **kwargs):
                 pl_vmax = pars['vmax']
                 del pars['vmax']
 
-            pl.imshow(self.source,cmap=pl_cmap,vmin=pl_vmin,vmax=pl_vmax)
-            pl.plot(self.xypos[0]-1,self.xypos[1]-1,pars['marker'])
+            plt.imshow(self.source,cmap=pl_cmap,vmin=pl_vmin,vmax=pl_vmax)
+            plt.plot(self.xypos[0]-1,self.xypos[1]-1,pars['marker'])
+            plt.figure() # create new figure for subsequent plots (by user?)
```

Comment:
Shouldn't figure() be called before plot()? CC: @sosey

Reply:
It might be better to work with your own figure so that users plots aren't being clobbered. I would recommend considering the following changes:

- move the import of matplotlib to the top of the module, into a try that sets something like `matplotlib_installed = True`; available for both classes
-  add a useful ImportError statement in the except block that tells the user matplotlib couldn't be found and plotting wont happen
- each catalog class can then have an instance figure attribute, I don't believe this will interfere with any other plots that the user has already, or will later, create
- now use the axes class call in matplotlib for creating the plot, this would make plotXYCatalog look something like the following:
```
def plotXYCatalog(self, **kwargs):
        """
        Method which displays the original image and overlays the positions
        of the detected sources from this image's catalog.
        Plotting `kwargs` that can be provided are:
            vmin, vmax, cmap, marker
        Default colormap is `summer`.
        """
 
        if matplotlib_installed:
            self.fig.clf()
            self.fig.add_subplot(111)
            ax = self.fig.gca()
            pars = kwargs.copy()
 
            if 'marker' not in pars:
                pars['marker'] = 'b+'
 
            if 'cmap' in pars:
                pl_cmap = pars['cmap']
                del pars['cmap']
            else:
                pl_cmap = 'summer'
            pl_vmin = None
            pl_vmax = None
            if 'vmin' in pars:
                pl_vmin = pars['vmin']
                del pars['vmin']
            if 'vmax' in pars:
                pl_vmax = pars['vmax']
                del pars['vmax']
 
            ax.imshow(self.source, cmap=pl_cmap, vmin=pl_vmin, vmax=pl_vmax)
            ax.plot(self.xypos[0]-1, self.xypos[1]-1, pars['marker'])
            self.fig.canvas.draw() # or plt.draw()
    else:
        print("Warning: Matplotlib not found, unable to plot")
 ```
draw() will refresh the plot in interactive mode if you are using object method and function calls.